### PR TITLE
BZ862325: Added tests for improper rule deactivation

### PR DESCRIPTION
**Pull request by winklerm**
Added 3 test cases for Bugzilla issue 862325: one for fact modification, the other two for fact retraction and insert. The test cases do not pass (including the one for fact modification which should be already fixed according to Bugzilla comment).

Please review the proposed tests, either they are not correct, or the fix to 862325 is not yet complete.

### CODE REVIEW

File: drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+// MODIFICATION
+
+// this rule must fire before the other two
+rule "rule1-modify"
+when
+    ActivationTestKind( kind == Kind.MODIFY )
+    $fact : ActivationTestFact( value == "default" )
+then
+    insert( "first_fired" );
```
Comment (esteban-aliverti) [inline drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl:15]: This insert() creates an Activation for rule2-modify and rule3-modify

File: drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    modify( $fact ) { setValue( "non-default" ) }
```
Comment (esteban-aliverti) [inline drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl:16]: This line here cancels the activations of rule2-modify and rule3-modify. So for the modify test, only rule1-modify should be fired after fireAllRules(). Is that not the case?

File: drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl (hunk 1 above)
Comment (esteban-aliverti) [inline drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl:15]: These activations are not going to be executed until the current activation's execution ends

File: drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl (hunk 2 above)
Comment (winklerm) [inline drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl:16]: Please have a look at https://bugzilla.redhat.com/show_bug.cgi?id=862325. I have understood that rules rule2-modify and rule3-modify should fire in this case. Did I get it right?

Comment (esteban-aliverti) [inline drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl:16]: Well, we will have to ask @etirelli. But AFAIK, the expected behavior here is that only 'rule1-modify' gets executed.

Comment (winklerm) [inline drools-compiler/src/test/resources/org/drools/integrationtests/activation/activation.drl:16]: I have asked @etirelli on irc and you are right, thanks for your comments!

**Pull request closed without merging**